### DISCUSSION TITLE
Any workaround to reduce AppImage size?

### DISCUSSION THREAD

**Discussion by momobobe** [tauri-apps/tauri#6918] [Q&A]
I found most AppImage files much larger than other packages like those for Windows, MacOS, or even other Linux packages. I see this is mentioned in docs, but still I wonder if there is any chance to debloat it (I guess it's caused by webkit2gtk, which should be safe to remove for most distros). Otherwise, there is no big difference from an Electron version in package size.

Or is there any other distro-agnostic format with a typically small packaging size supported now?

**Comment by FabianLars** [comment]
No there is no way, well, i guess a few KB here and there in case the bundler includes a few things too eagerly.

> I guess it's caused by webkit2gtk, which should be safe to remove for most distros

Not at all, even though most Gnome based distros do have it installed you can't really expect this to be the case, because if it's not there the appimage can't handle that in contrast to something like the `.deb` bundle which installs it on the system if needed.

At the end of the day this is just how appimages work, they are supposed to bundle everything they need to run and not doing this would cause more harm than good.

> Or is there any other distro-agnostic format with a typically small packaging size supported now?

Well, i don't think there even is something that matches that description. _Small_ formats, like .deb, .rpm, and whatever are repo specific, and distro-agnostic formats like flatpak will install large runtimes on the system to function, so if you count the runtime size to the app bundle size then these are also not small at all.

This is really something the Linux ecosystem has to "solve".

**Comment by ttntm** [comment]
Just gave Tauri a shot and built an AppImage for my [time tracker](https://github.com/ttntm/itrack). It's a small Vue SPA - the web build amounts to 647.9 kB and the whole `node_modules` folder is 120.2 MB in size (36.2 MB of this is the folder `@tauri-apps`).

The resulting AppImage is 89.3 MB which I find quite extreme TBH.

The Vite build is optimized to the best of my knowledge and the ~650 kB of the web build are ok IMO. 

I've also read through [Reducing App Size](https://tauri.app/v1/guides/building/app-size); `cargo-bloat` didn't show anything noteworthy, here's the output:

```txt
File  .text    Size              Crate Name
 0.8%   3.2% 77.7KiB             x11_dl x11_dl::xlib::Xlib::open::{{closure}}
 0.8%   3.1% 75.4KiB  tauri_runtime_wry tauri_runtime_wry::handle_user_message
 0.4%   1.7% 41.5KiB  tauri_runtime_wry tauri_runtime_wry::handle_user_message
 0.2%   0.9% 22.5KiB                std std::backtrace_rs::symbolize::gimli::Cache::with_global
 0.2%   0.7% 18.1KiB                std std::backtrace_rs::symbolize::gimli::Context::new
 0.2%   0.7% 16.4KiB              tauri tauri::manager::WindowManager<R>::prepare_window
 0.1%   0.6% 14.1KiB tauri_runtime_wry? <tauri_runtime_wry::Wry<T> as tauri_runtime::Runtime<T>>::create_window
 0.1%   0.5% 12.8KiB               glib glib::main_context_channel::dispatch
 0.1%   0.5% 11.5KiB                std addr2line::ResUnit<R>::find_function_or_location::{{closure}}
 0.1%   0.5% 11.0KiB                wry wry::webview::webkitgtk::InnerWebView::new
21.5%  85.3%  2.0MiB                    And 7588 smaller methods. Use -n N to show more.
25.2% 100.0%  2.4MiB                    .text section size, the file size is 9.4MiB
```

Looks nothing like ~90 MB to me.

Any ideas what else I could do to reduce the AppImage file's size, or is this just how it is?

NB: based on [this Baeldung article](https://www.baeldung.com/linux/snaps-flatpak-appimage#3-application-size), the Discord AppImage is 98 MB and that's not a simple <1 MB time tracker...

**Reply by FabianLars** [reply]
> Any ideas what else I could do to reduce the AppImage file's size, or is this just how it is?

It is what it is. AppImages, _by design_ include (almost) all the system dependencies they need to run.

> NB: based on [this Baeldung article](https://www.baeldung.com/linux/snaps-flatpak-appimage#3-application-size), the Discord AppImage is 98 MB and that's not a simple <1 MB time tracker...

Yeah, but the discord code itself is probably also just 10mb max, the rest is all the libraries and runtimes, like in discord's case the biggest one would be chromium, and in tauri it's webkitgtk.
If we count into the fact that the appimage is heavily compressed (compare it with the size of the .AppDir folder next to it) the size of your app executable doesn't really affect its size that much. As-in depending on what causes your app binary to increase in size, it may not even be measurable on the resulting appimage.

**Reply by ttntm** [reply]
Ok, thanks.

I'll keep digging a bit more and accept AppImage for what it is otherwise :)
